src/docs/testing/integration-tests/index.md

```diff
@@ -0,0 +1,285 @@
+Flutter widgets without running the app itself. An **integration test** runs the
+full app on the target device and reports the results back to the host machine.
+
+The flutter_driver package runs integration tests written in Dart on a target
+device and reports the results to the host.
+
+The [integration_test][] package lets you run tests directly on the target
+device. This allows you to run integration tests on multiple Android or iOS
+devices using Firebase TestLab. These tests can also be run using
+flutter_driver.
+
+## Project Setup
+
+Add `integration_test`, `flutter_test`, and optionally `flutter_driver` to your
+pubspec.yaml file:
+
+```yaml
+integration_test: ^0.9.0
+flutter_test:
+  sdk: flutter
+flutter_driver:
+  sdk: flutter
+```
+
+Create a new directory `integration_test/` with a new file,
+`<name>_test.dart`:
+
+```dart
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  testWidgets("failing test example", (WidgetTester tester) async {
+    expect(2 + 2, equals(5));
+  });
+}
+```
+
+{{site.alert.note}}
+Note: You should only use `testWidgets` to declare your tests, or errors will
+not be reported correctly.
+{{site.alert.end}}
+
+
+To run tests with `flutter drive`, create a new directory containing a new file,
+`test_driver/integration_test_driver.dart`:
+
+```dart
+import 'package:integration_test/integration_test_driver.dart';
+
+Future<void> main() => integrationDriver();
+```
+
+### Directory Structure
+
+```
+lib/
+  ...
+integration_test/
+  foo_test.dart
+  bar_test.dart
+test/
+  # Other unit tests go here.
+test_driver/
+  integration_test.dart
+```
+
+See also:
+
+ * [integration_test usage][]
+
+## Running Using Flutter Driver
+
+These tests can be launched with the `flutter drive` command, where
+`<DEVICE_ID>`: is the optional device ID or pattern displayed in the output of the
+`flutter devices` command:
+
+```bash
+flutter drive \
+  --driver=test_driver/integration_test.dart \
+  --target=integration_test/foo_test.dart \
+  -d <DEVICE_ID>
+```
+
+This runs the tests in `foo_test.dart` via the adapter in
+`test_driver/integration_test.dart`.
+
+
+### Running in a browser 
+
+First, [Download and install ChromeDriver][] and run it on port 4444:
+
+
+```
+chromedriver --port=4444
+```
+
+In a separate process, run `flutter_drive`:
+
+```
+flutter drive \
+  --driver=test_driver/integration_test_driver.dart \
+  --target=integration_test/counter_test.dart \
+  -d web-server
+```
+
+## Testing on Firebase TestLab 
+
+
+### Android Setup
+Follow the instructions in the [Android Device Testing][] section of the README.
+
+### iOS Setup
+Follow the instructions in the [iOS Device Testing][] section of the README.
+
+### TestLab Project Setup
+
+Go to the [Firebase Console], and create a new project if you don't have one
+already. Then navigate to Quality > Test Lab:
```

Comment:
Menu items should be in bold, so `**Quality > Test Lab**`.   https://developers.google.com/style/ui-elements?hl=en